modules/contact/src/actions/ExplicitDynamicsContactAction.C

```diff
@@ -0,0 +1,442 @@
+  // system is solved to obtain accelerations (a = M^{-1}F) such that only velocities on the
+  // contacting boundary need to be updated with the contact algorithm output.
+  mooseWarning("Verification of explicit dynamics capabilities is an ongoing effort.");
+}
+
+void
+ExplicitDynamicsContactAction::act()
+{
+  // proform problem checks/corrections once during the first feasible task
+  if (_current_task == "add_contact_aux_variable")
+  {
+    if (!_problem->getDisplacedProblem())
+      mooseError(
+          "Contact requires updated coordinates.  Use the 'displacements = ...' parameter in the "
+          "Mesh block.");
+
+    // It is risky to apply this optimization to contact problems
+    // since the problem configuration may be changed during Jacobian
+    // evaluation. We therefore turn it off for all contact problems so that
+    // PETSc-3.8.4 or higher will have the same behavior as PETSc-3.8.3.
+    if (!_problem->isSNESMFReuseBaseSetbyUser())
+      _problem->setSNESMFReuseBase(false, false);
+  }
+
+  addNodeFaceContact();
+
+  if (_current_task == "add_aux_kernel")
+  { // Add ContactPenetrationAuxAction.
+    if (!_problem->getDisplacedProblem())
+      mooseError("Contact requires updated coordinates.  Use the 'displacements = ...' line in the "
+                 "Mesh block.");
+    // Create auxiliary kernels for each contact pairs
+    for (const auto & contact_pair : _boundary_pairs)
+    {
+      {
+        InputParameters params = _factory.getValidParams("PenetrationAux");
+        params.applyParameters(parameters(),
+                               {"secondary_gap_offset", "mapped_primary_gap_offset", "order"});
+
+        std::vector<VariableName> displacements =
+            getParam<std::vector<VariableName>>("displacements");
+        const auto order = _problem->systemBaseNonlinear(/*nl_sys_num=*/0)
+                               .system()
+                               .variable_type(displacements[0])
+                               .order.get_order();
+
+        params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{order});
+        params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR};
+        params.set<std::vector<BoundaryName>>("boundary") = {contact_pair.second};
+        params.set<BoundaryName>("paired_boundary") = contact_pair.first;
+        params.set<AuxVariableName>("variable") = "penetration";
+        if (isParamValid("secondary_gap_offset"))
+          params.set<std::vector<VariableName>>("secondary_gap_offset") = {
+              getParam<VariableName>("secondary_gap_offset")};
+        if (isParamValid("mapped_primary_gap_offset"))
+          params.set<std::vector<VariableName>>("mapped_primary_gap_offset") = {
+              getParam<VariableName>("mapped_primary_gap_offset")};
+        params.set<bool>("use_displaced_mesh") = true;
+        std::string name = _name + "_contact_" + Moose::stringify(ed_contact_auxkernel_counter++);
```

Comment:
I see you are using the action name. Then the counter could be a local variable. Ne need for it to be a global static.